pkg/apis/azure/v1alpha1/types_worker.go

```diff
@@ -19,8 +19,13 @@ type WorkerConfig struct {
 	// NodeTemplate contains resource information of the machine which is used by Cluster Autoscaler to generate nodeTemplate during scaling a nodeGroup from zero.
 	// +optional
 	NodeTemplate *extensionsv1alpha1.NodeTemplate `json:"nodeTemplate,omitempty"`
-	// DiagnosticsProfile specifies boot diagnostic options
+
+	// DiagnosticsProfile specifies boot diagnostic options.
```

Comment:
since you went into the trouble to add the new line, please also add the +optional here